app/redux/ducks/interventions.js

```diff
@@ -46,43 +50,81 @@ export function unsubscribe(channel) {
   return { type: UNSUBSCRIBE, payload: channel };
 }
 
-export function notify(notification) {
-  const { message } = notification.data;
-  const { type } = message;
-  if (type === 'subject_queue') {
-    const subjectIDs = message.subject_ids;
-    const workflowID = message.workflow_id;
-    return (dispatch) => {
+
+function prependSubjectQueue(data) {
+  const subjectIDs = data.subject_ids;
+  const workflowID = data.workflow_id;
+  return (dispatch) => {
+    dispatch({
+      type: FETCH_SUBJECTS,
+      payload: subjectIDs
+    });
+    apiClient.type('subjects').get(subjectIDs)
+    .catch((error) => {
       dispatch({
-        type: FETCH_SUBJECTS,
-        payload: subjectIDs
-      });
-      apiClient.type('subjects').get(subjectIDs)
-      .catch((error) => {
-        dispatch({
-          type: ERROR,
-          payload: error
-        });
-        return [];
-      })
-      .then(subjects => subjects.map((subject) => {
-        subject.update({ 'metadata.intervention': true });
-        return subject;
-      }))
-      .then((subjects) => {
-        dispatch(prependSubjects(subjects, workflowID));
+        type: ERROR,
+        payload: error
       });
-    };
-  } else {
-    return { type: ADD_NOTIFICATION, payload: notification };
+      return [];
+    })
+    .then(subjects => subjects.map((subject) => {
+      subject.update({ 'metadata.intervention': true });
+      return subject;
+    }))
+    .then((subjects) => {
+      dispatch(prependSubjects(subjects, workflowID));
+    });
+  };
+};
+
+export function intervention(message) {
+  // Example payload data from sugar
+  // {
+  //  channel: "user:27"
+  //    data: {
+  //      event: "intervention",
+  //      event_type: "message",
+  //      message: "You are a star! All of your contributions really help.",
+  //      project_id: "908",
+  //      type: "experiment",
+  //      user_id: "27",
+  //    }
+  //  type: "experiment"
+  // }
+
+  // Only process known intervention experiment events from sugar.
+  var { type = 'unknown' } = message;
+  if (type !== "experiment") {
+    return { type: UNKNOWN_EXPERIMENT };
```

Comment:
Rather than defining a new action type for each error, you could define a single `INTERVENTION_ERROR` action type, and pass the error details in the action payload. eg. `{ type: INTERVENTION_ERROR, payload: "No data found in message." }`. I think it depends whether you want each type of error to have its own handler or not. If each error makes the same change to Redux state, then one action type is probably sufficient.

Reply:
good idea - i'll roll them up to use the ERROR action with different payload messages describing the different states.